states.h

```diff
@@ -16,48 +17,50 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#ifndef NFLUA_STATES_H
-#define NFLUA_STATES_H
+#ifndef KLUA_STATES_H
+#define KLUA_STATES_H
 
-#include "kpi_compat.h"
+#include "lua/lua.h"
 
-#include <lua.h>
+#define KLUA_NAME_MAXSIZE 64
+#define KLUA_MIN_ALLOC_BYTES (32 * 1024UL)
+#define KLUA_MAX_BCK_COUNT 32
 
-#define NFLUA_NAME_MAXSIZE 64
-
-struct xt_lua_net;
+struct meta_state {
+	struct hlist_head states_table[KLUA_MAX_BCK_COUNT];
+	spinlock_t statestable_lock;
+	spinlock_t rfcnt_lock;
+	atomic_t states_count;
+};
 
-struct nflua_state {
+struct klua_state {
 	struct hlist_node node;
 	lua_State *L;
-	struct xt_lua_net *xt_lua;
 	spinlock_t lock;
-	kpi_refcount_t users;
+	refcount_t users;
+	#ifndef LUNATIK_UNUSED
 	u32 dseqnum;
+	#endif /*LUNATIK_UNUSED*/
 	size_t maxalloc;
 	size_t curralloc;
-	unsigned char name[NFLUA_NAME_MAXSIZE];
+	unsigned char name[KLUA_NAME_MAXSIZE];
 };
 
+#ifndef LUNATIK_UNUSED
 typedef int (*nflua_state_cb)(struct nflua_state *s, unsigned short *total);
+#endif /*LUNATIK_UNUSED*/
 
-struct nflua_state *nflua_state_create(struct xt_lua_net *xt_lua,
-        size_t maxalloc, const char *name);
-
-int nflua_state_destroy(struct xt_lua_net *xt_lua, const char *name);
-
-struct nflua_state *nflua_state_lookup(struct xt_lua_net *xt_lua,
-        const char *name);
+struct klua_state *klua_state_create(size_t maxalloc, const char *name);
+int klua_state_destroy(const char *name);
+struct klua_state *klua_state_lookup(const char *name);
 
+#ifndef LUNATIK_UNUSED
 int nflua_state_list(struct xt_lua_net *xt_lua, nflua_state_cb cb,
 	unsigned short *total);
+#endif /*LUNATIK_UNUSED*/
 
-void nflua_state_destroy_all(struct xt_lua_net *xt_lua);
-
-bool nflua_state_get(struct nflua_state *s);
-void nflua_state_put(struct nflua_state *s);
-
-void nflua_states_init(struct xt_lua_net *xt_lua);
```

Comment:
This function isn't here anymore because it will not be exported to the user (i.e is used only on file `lunatik_core.c` to access and initialize the static variable `static struct meta_state ms` and its dependencies)